### ISSUE TITLE
Handle multiple functions triggering the same output component

### ISSUE THREAD

**Issue by aliabid94** [gradio-app/gradio#1448]
Have to figure out how to properly deal with this, both in queue and nnon-queue scenarios.

**Comment by FarukOzderim** [comment]
We might also have a `sequential_events` parameter in launch, which will ignore all the event triggers while the demo is waiting for an event to return from the queue.

**Comment by abidlabs** [comment]
@aliabid94 this issue is unclear to me from the description. Is it still an issue or can we close?

**Comment by FarukOzderim** [comment]
Well the question was this,


event 1 : 
input: a,b output: c

event 2: 
input b,c output: c

What happens if one triggers them one after another?

**Comment by abidlabs** [comment]
Hmm well here's a simple MVP:

```py
import gradio as gr
import time

def first(x,y):
    time.sleep(4)
    return x

with gr.Blocks() as demo:
    a = gr.Textbox()
    b = gr.Textbox()
    c = gr.Textbox()
    
    b1 = gr.Button()
    b2 = gr.Button()
    
    b1.click(first, [a,b], c)
    b2.click(first, [b,c], c)
    
demo.launch()
```

If you click on `b1` and then `b2`, what ends up happening is that the output component keeps the loading indicator running until both events finish running:

<img width="570" alt="image" src="https://user-images.githubusercontent.com/1778297/193334424-c3a5d0e4-3fb4-446b-96f0-b567b8a4c93d.png">

That's what I would expect to happen, so I'm not sure what the underlying issue is?

**Comment by FarukOzderim** [comment]
Well back then we discussed the issue with @aliabid94 in length, and didn't arrive at a conclusion, so I think he can explain it better in an online meeting.

Basically, this makes it a complex situation, and there are several ways to execute them.

For example right now you never see the output of the first event when they are executed one after another.

```
!pip install gradio -q

import gradio as gr
import time

def first(x,y):
    time.sleep(4)
    return x

def second(x,y):
    time.sleep(2)
    return x+y

with gr.Blocks() as demo:
    a = gr.Textbox()
    b = gr.Textbox()
    c = gr.Textbox()
    
    b1 = gr.Button()
    b2 = gr.Button()
    
    b1.click(first, [a,b], c)
    b2.click(second, [b,c], c)
    
demo.launch()
```

Several ways of execution:
- Do not accept the event when there are input or output components that are currently in processing 
- Execute the event directly with what you have
- Wait and execute the event automatically after the waiting components are processed. 

Btw this is not a big problem, just need to make it explicit about how it works, and select a strategy.

**Comment by abidlabs** [comment]
Ah got it, thanks for clarifying @FarukOzderim. I'm not sure if there is a strong need to change the current way (which is basically to overwrite the first event's output with the second event) but we can be more explicit about it in the documentation.

**Comment by abidlabs** [comment]
This will be addressed by #1954 in 4.0. Closing for now, thanks @omerXfaruq and @aliabid94!
